include/EVT/io/SPI.hpp

```diff
@@ -0,0 +1,151 @@
+#ifndef _EVT_SPI_
+#define _EVT_SPI_
+
+#include "GPIO.hpp"
+#include <stdint.h>
+
+#define SPI_MSB_FIRST 1
+#define SPI_LSB_FIRST 0
+
+#define SPI_MODE0 0x01
+#define SPI_MODE1 0x02
+#define SPI_MODE2 0x04
+#define SPI_MODE3 0x08
+
+#define SPI_SPEED_4MHZ 4000000
+#define SPI_SPEED_2MHZ 2000000
+#define SPI_SPEED_1MHZ 1000000
+#define SPI_SPEED_500KHZ 500000
+#define SPI_SPEED_250KHZ 250000
+#define SPI_SPEED_125KHZ 125000
+#define SPI_SPEED_62KHZ 62500
+#define SPI_SPEED_31KHZ 31250
+
+namespace EVT::core::IO {
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+enum class Pin;
+
+class SPI {
+public:
+    /**
+     * Constructs an SPI instance in full duplex mode to send and receive data
+     * @param CSPins an array of chip select pins for selecting which device to communicate with.
+     * @param pinLength the number of pins in the chip select array
+     * @param sckPin the pin for the clk line
+     * @param mosiPin the mosi pin for sending data
+     * @param misoPin the miso pin for receiving data
+     */
+    SPI(GPIO* CSPins[], uint8_t pinLength, Pin sckPin, Pin mosiPin, Pin misoPin);
+
+    /**
+     * Constructs an SPI instance in half duplex mode to only send data
+     * @param CSPins an array of chip select pins for selecting which device to communicate with.
+     * @param pinLength the number of pins in the chip select array
+     * @param sckPin the pin for the clk line
+     * @param mosiPin the mosi pin for sending data
+     */
+    SPI(GPIO* CSPins[], uint8_t pinLength, Pin sckPin, Pin mosiPin);
+
+    /**
+     * begin a device transmission. Call before each set of read and write interactions.
```

Comment:
NIT: The block comments are normally capitalized. It's probably fine if you leave them lowercase if you make them consistent. There's also normally an empty line between the explanation of the method and the parameters/return section of the comment.